### PR TITLE
Update Android dependencies

### PR DESCRIPTION
**Pull request by ThexXTURBOXx**
Since JCenter is abandoned, it might make sense to replace it with Maven Central and just update the dependencies in general.

The only hacky part is the now-needed null-safe operator in the Kotlin Plugin. Although, in the past, this was also internally never a problem. So, I guess, it is safe to assume that `context` will not be `null` at this point.

This PR would be very nice to see in 0.3.1 as well :)

### CODE REVIEW

**Comment by LinusU** [comment]
Sorry, missed this before cutting 0.3.1, but we can just release a 0.3.2 if need be.

I have a few quick questions though:

1. Is changing the `compileSdkVersion` from 28 to 31 a breaking change? Will it affect the target platform of any application using this library?
2. Is there a specific bug/issue that this fixes, or is it just good cleanup?

**Comment by ThexXTURBOXx** [comment]
No problem! :)
1. It's not really a breaking change (i.e. it doesn't break compatibility with older Android releases), but rather a future-proof thing to do. This way, we can ensure, the API we are accessing is still available and (hopefully) not deprecated.
2. I noticed some compatibility problems when another library is using `androidx.browser` as well and there is a version mismatch between these. Then, one overrides the other and there are clashes and errors (like the missing `!`)

**Comment by ThexXTURBOXx** [comment]
Done. All dependencies are now up-to-date (I updated the example as well now) and I fixed the example as well.
This PR is now ready to get merged.

File: example/lib/main.dart [hunk 1]
```diff
@@ -91,7 +91,7 @@ class _MyAppState extends State<MyApp> {
   }
 
   void authenticate() async {
-    final url = 'http://localtest.me:43823/';
+    final url = 'http://127.0.0.1:43823/';
```
Comment (LinusU) [inline example/lib/main.dart:94]: Why was this changed? 🤔

Comment (ThexXTURBOXx) [reply example/lib/main.dart:94]: For me, `localtest.me` appearently doesn't work anymore (at least, it didn't work yesterday)

Comment (ThexXTURBOXx) [reply example/lib/main.dart]: Just checked again. `localtest.me` doesn't work in Android Emulators. Haven't tested on real devices yet.
However, since most testing is done on Android Emulators anyways, I would leave it as `localhost` now.
If you think, I should change it back, then I will do that :)
Otherwise, this PR is ready for merging

**Comment by ThexXTURBOXx** [comment]
@LinusU Any news on this? :)

File: example/lib/main.dart [hunk 2]
```diff
@@ -55,7 +55,7 @@ const html = """
 <body>
   <main>
     <div id="icon">&#x1F3C7;</div>
-    <div id="text">Press the button below to sign in using your Localtest.me account.</div>
+    <div id="text">Press the button below to sign in using your Localhost account.</div>
```
Comment (LinusU) [inline example/lib/main.dart:58]: I think that this is better suited for a separat PR so that we can update the screenshots, and make sure that this works on all platforms...

```suggestion
    <div id="text">Press the button below to sign in using your Localtest.me account.</div>
```

File: example/lib/main.dart [hunk 3]
```diff
@@ -91,7 +91,7 @@ class _MyAppState extends State<MyApp> {
   }
 
   void authenticate() async {
-    final url = 'http://localtest.me:43823/';
+    final url = 'http://localhost:43823/';
```
Comment (LinusU) [inline example/lib/main.dart:94]:
```suggestion
    final url = 'http://localtest.me:43823/';
```

**Comment by LinusU** [comment]
Sorry for the long delay here!

This is now released as [🚢 0.4.0 / 2021-12-11](https://github.com/LinusU/flutter_web_auth/releases/tag/0.4.0)
